src/Ombi.Core/Engine/TvRequestEngine.cs

```diff
@@ -608,9 +609,49 @@ private async Task<RequestEngineResult> AfterRequest(ChildRequests model)
                 RequestDate = DateTime.UtcNow,
                 RequestId = model.Id,
                 RequestType = RequestType.TvShow,
+                EpisodeCount = model.SeasonRequests.Select(m => m.Episodes.Count).Sum(),
             });
 
             return new RequestEngineResult { Result = true };
         }
+
+        public async Task<RequestQuotaCountModel> GetRemainingRequests()
+        {
+            OmbiUser user = await GetUser();
+            int limit = user.EpisodeRequestLimit ?? 0;
+
+            if (limit <= 0)
+            {
+                return new RequestQuotaCountModel()
+                {
+                    HasLimit = false,
+                    Limit = 0,
+                    Remaining = 0,
+                    NextRequest = DateTime.Now,
+                };
+            }
+
+            IQueryable<RequestLog> log = _requestLog.GetAll().Where(x => x.UserId == user.Id && x.RequestType == RequestType.TvShow);
+
+            // Needed, due to a bug which would cause all episode counts to be 0
+            int zeroEpisodeCount = await log.Where(x => x.EpisodeCount == 0).Select(x => x.EpisodeCount).CountAsync();
```

Comment:
Wait, why do we do this?

Reply:
See #2464 for the bug I'm referring to.

Prior to this PR, we weren't saving the `EpisodeCount` to the logs, which means that all of the old log entries will all have `EpisodeCount` set to 0. 

This query gets around this issue by counting all of these instances and assuming that they have 1 episode assigned to them.

On a side note, looks like I'm searching through all requests, rather than just requests in the past week. Will fix this and check that I haven't done it elsewhere.